data/battle_scripts_1.s

```diff
@@ -20,6 +20,36 @@
 
 	.section script_data, "aw", %progbits
 
+BattleScript_EffectSpicyExtract::
+	attackcanceler
+	jumpifsubstituteblocks BattleScript_ButItFailed
+	accuracycheck BattleScript_ButItFailed, NO_ACC_CALC_CHECK_LOCK_ON
+	attackstring
+	ppreduce
+	setstatchanger STAT_ATK, 2, FALSE
+	statbuffchange STAT_CHANGE_ALLOW_PTR, BattleScript_EffectSpicyExtractDefenseDown2
+	jumpifbyte CMP_EQUAL, cMULTISTRING_CHOOSER, B_MSG_STAT_WONT_INCREASE, BattleScript_EffectSpicyExtractDefenseDown2
```

Comment:
Test this against a Pokemon with Contrary holding a Clear Amulet.

Reply:
What should be happening in this case? My intuition said that contrary should boost defense by 2 stages (this is happening with the current implementation) but testing on showdown it seems that the attack is raised instead and no defense drop. If showdown is correct, then it seems like Clear Amulet overrides Contrary right?

Reply:
Sounds like it could be a bug with Showdown. Worth checking the correct implementation in SV directly.

Reply:
My bad. I tested again and it seems I forgot to change the ability on Snivy on showdown 🫣

It might still be a good idea to test it in SV because I assume this order is wrong:

https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/a7279984-aa45-4956-8f85-75d8aa737990

It should probably be attackanimation, clear amulet, stat change.

Reply:
The above clip was obviously wrong (@AsparagusEduardo tested in on SV).

https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/e531ae74-32cd-488c-8af4-8b23119c6370

@Skeli789 I've push a fix but it looks a bit hacky. If you think there is a better way to do it, I'll see if I can find it. 
Also some simple AI additions (needs testing though)

Reply:
Here's the recorded video in question
https://www.youtube.com/watch?v=QSqKshwGPgI&ab_channel=AsparagusEduardo

Reply:
It would be less hacky to  make a helper function called something like `DoesSpicyExtractHaveNoEffect`, and then if that is true, then you can set the result flags to no effect and the animation will automatically be skipped. If the script is set up correctly the correct strings will also print at each stat buff command.

Reply:
Sorry, it took so long but thanks to the hint I was able to simplify the script a little bit more. Though I couldn't figure out how to rework the small hack for the clear amulet / contrary interaction.

Reply:
Sent in DM